### DISCUSSION TITLE
What import should I used when Embedding?

### DISCUSSION THREAD

**Discussion by andykillen** [roadrunner-server/roadrunner#1971] [Q&A] [answered]
I feel like I am running in circles on trying to work out what import's I should be using to allow for Embedding roadrunner? 

example go code:
```
overrides := []string{} // List of configuration overrides
plugins := roadrunner.DefaultPluginsList() // List of RR plugins to enable
rr, err := roadrunner.NewRR(".rr.yaml", overrides, plugins)
```

I've tried 
- github.com/spiral/roadrunner, does not have the functions .NewRR() and .DefaultPluginsList() so fails, its just got types and no functions
- github.com/roadrunner-server/roadrunner, but it only redirects to or complains about the path being declared as  github.com/spiral/roadrunner
- github.com/roadrunner-server/roadrunner/v2024, has the needed functions do exits, but when trying to go get it, there is no error, but with go mod tidy I get

```
go mod tidy
go: finding module for package github.com/roadrunner-server/roadrunner/v2024
go: custom_connector/client imports
        github.com/roadrunner-server/roadrunner/v2024: module github.com/roadrunner-server/roadrunner/v2024@latest found (v2024.1.5), but does not contain package github.com/roadrunner-server/roadrunner/v2024
```
So of course I have tried go get github.com/roadrunner-server/roadrunner/v2024@latest, but the same message stands.

No amount of clearing mod cache, running mod vendor or mod tidy between attempts of go get make any different to the situation that I can see. 
        
How should the import be setup to give me access to roadrunner.NewRR(), roadrunner.DefaultPluginList(), etc ? and what go get command should I be using?

**Comment by rustatian** [comment, marked as answer]
Hey @andykillen 👋 
You should use `"github.com/roadrunner-server/roadrunner/v2024/lib"` and use `lib.NewRR()` to start 😃

**Reply by andykillen** [reply]
many thanks for the reply. Looks to have done the trick.  :)

**Reply by rustatian** [reply]
I'll update the docs to make it more clear 😃
